[Ashley Goh] iP

ashleygoh1 opened the pull request:
### Bee

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

**Features:** 
- [ ] 3 different type of task 
  - _Todo_ 
  - Deadlines
  - Event
- [ ] Managing deadlines (coming soon)
- [ ] Reminder (coming soon)

Try out by
1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/369)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😄 

If you are a Java programmer, you can use it to practice Java too. Here is the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
change logo and added message

Commit:
add echo

Commit:
add list

Commit:
add horizontal lines

Commit:
add indent, task class, and checkbox feature

Commit:
add indent, task class, and checkbox feature

Commit:
add Todos, Events and Deadlines class

Commit:
add indent, task class, and checkbox feature

Commit:
Update test file

Commit:
handle exceptions

Commit:
delete task

Commit:
handle exceptions

ashleygoh1 committed:
Merge pull request #1 from ashleygoh1/Level-0

change logo and added message

ashleygoh1 committed:
Merge pull request #2 from ashleygoh1/Level-1

level-1: add echo

ashleygoh1 committed:
Merge branch 'master' into Level-3

ashleygoh1 committed:
Merge pull request #3 from ashleygoh1/Level-3

Level-3: add indent, list, task class, and checkbox feature

ashleygoh1 committed:
Revert "Level-3: add indent, list, task class, and checkbox feature"

ashleygoh1 committed:
Merge pull request #4 from ashleygoh1/revert-3-Level-3

Revert "Level-3: add indent, list, task class, and checkbox feature"

ashleygoh1 committed:
Merge branch 'master' into Level-4

ashleygoh1 committed:
Merge pull request #5 from ashleygoh1/Level-4

add Todos, Events and Deadlines class

ashleygoh1 committed:
Merge pull request #6 from ashleygoh1/A-TextUiTesting

Update test file

ashleygoh1 committed:
Merge pull request #7 from ashleygoh1/Level-5

Level 5: handle exceptions

ashleygoh1 committed:
Merge pull request #8 from ashleygoh1/Level-6

Level-6: delete task

Commit:
remove last space from the initial message

Commit:
add multiple exception files, read from file, write into file

Commit:
add multiple exception files, read from file, write into file

Commit:
change variable datetime from type String to LocalDate and LocalTime

ashleygoh1 committed:
Merge branch 'master' into branch-Level-7

ashleygoh1 committed:
Merge pull request #9 from ashleygoh1/branch-Level-7

Branch level 7 - Save

ashleygoh1 committed:
Merge pull request #10 from ashleygoh1/branch-Level-8

Level 8 - Dates and Times

Commit:
Add more classes to make code more OOP

ashleygoh1 committed:
Merge branch 'master' into A-MoreOOP

ashleygoh1 committed:
Merge pull request #11 from ashleygoh1/A-MoreOOP

A-Packages

Commit:
put files into different packages

ashleygoh1 committed:
Merge pull request #12 from ashleygoh1/A-Packages

A-Packages

ashleygoh1 committed:
Merge pull request #13 from ashleygoh1/add-gradle-support

Add gradle support

Commit:
add test cases for Task (todo, deadline, event)

Commit:
Create a JAR File

Commit:
Add JavaDoc and edited the DateTime format

Commit:
Add JavaDoc comments

Commit:
Tweak the code to comply with a coding standard

Commit:
Give users a way to find a task by searching for a keyword

Commit:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

* branch-A-JavaDoc:
  Add JavaDoc comments
  Add JavaDoc and edited the DateTime format

# Conflicts:
#	src/main/java/command/byeCommand.java
#	src/main/java/command/deadlineCommand.java
#	src/main/java/command/deleteCommand.java
#	src/main/java/command/eventCommand.java
#	src/main/java/command/markCommand.java
#	src/main/java/command/showListCommand.java
#	src/main/java/command/todoCommand.java
#	src/main/java/command/unmarkCommand.java
#	src/main/java/duke/Storage.java

Commit:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  Tweak the code to comply with a coding standard
  Add JavaDoc comments
  Add JavaDoc and edited the DateTime format

# Conflicts:
#	data/tasks.txt
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Commit:
comment unused codes

ashleygoh1 committed:
Merge pull request #14 from ashleygoh1/branch-Level-9

Branch level 9

joenus commented on src/main/java/command/findCommand.java:
```diff
@@ -0,0 +1,33 @@
+    public findCommand(TaskList taskList, Ui ui) {
+        super(taskList, ui);
+    }
+    @Override
+
```
Might have added the spacing here in the wrong place.

joenus commented on src/main/java/command/markCommand.java:
```diff
@@ -0,0 +1,35 @@
+    public markCommand(TaskList taskList, Ui ui) {
+        super(taskList, ui);
+    }
+    @Override
+
+    public void execute(TaskList taskList, Ui ui) throws Exception {
```
Might have added the spacing here in the wrong place.

joenus commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+        this.time = time;
+    }
+
+    /**
+     * Returns the string representation of the task.
+     *
+     * @return String representation of the task.
+     */
```
Potentially unrequired.

joenus commented on src/test/java/task/EventTest.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Test
+    public void testToString() {
+        LocalDate startDate = LocalDate.of( 2022 , 2 , 2);
+        LocalTime startTime = LocalTime.of(10, 00);
+        LocalDate endDate = LocalDate.of( 2022 , 2 , 10);
```
Additional spacing.

```suggestion
        LocalDate startDate = LocalDate.of(2022 , 2, 2);
        LocalTime startTime = LocalTime.of(10, 00);
        LocalDate endDate = LocalDate.of(2022 , 2, 10);
```

joenus commented on src/test/java/task/DeadlineTest.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Test
+    public void testToString() {
+        LocalDate date = LocalDate.of(2022 , 2 , 2);
```
```suggestion
        LocalDate date = LocalDate.of(2022, 2, 2);
```

Additional spacing.

Commit:
Use checkStyle to detect coding style violations

Commit:
Add a GUI to the chatbot.
Add multiple files (MainWindow.fxml, DialogBox.fxml, images, Main.java, MainWindow.java, Launcher.java) to use JavaFX technology to implement the GUI.

Add a new WelcomeCommand.java that will show the welcome message when user type 'Hello'.

change the return type from void to String in TaskList.java.

ashleygoh1 committed:
Merge pull request #15 from ashleygoh1/branch-Level-10

Branch level 10

Commit:
Remove unnecessary line seperator and indentation

Update Javadoc comment

Commit:
Remove unnecessary line seperator and indentation

Update Javadoc comment

zengzihui commented on src/main/java/command/InvalidCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * @throws InvalidCommandException If input is not valid.
+     */
+    public InvalidCommand(TaskList taskList, Ui ui, Storage storage) {
+
```
Perhaps can remove the unnecessary line breaks

zengzihui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,226 @@
+     *
+     * @param t New todo to be added into taskList.
+     */
+    public String todo(Todo t) {
```
perhaps can use verbs for methods, something like addTodo(), to improve the readability.

zengzihui commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,141 @@
+                    String[] temp2 = temp.split("\\(from: ");
+                    String description = temp2[0];
+                    String result = temp2[1];
+                    LocalDate startDate = LocalDate.parse(result.substring(0, 11), dateTimeFormatter);
```
perhaps a LocalDateTime class can be used to store the start and end of an event since you want to keep track of both date and time

zengzihui commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,141 @@
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
```
Maybe can consider customizing the error messages for different types of errors to enhance the user experience, instead of solely relying on printStackTrace(). Besides that, I am not sure if the string parsing approach used here is a bit of hard coding, which could lead to maintenance issues if the structure of the input data changes. For example, if 'event' is modified to another word, all the substrings with specific indices need to be adjusted accordingly. 

One alternative approach is to use a delimiter, such as "|" to separate different fields in the input data. This would allow for easier parsing using methods like split() rather than relying on substrings with specific indices.

Commit:
Add a GUI to the chatbot
Add multiple files (MainWindow.fxml, DialogBox.fxml, images, Main.java, MainWindow.java, Launcher.java) to use JavaFX technology to implement the GUI

Remove unnecessary line seperator and indentation so that the output shown by the chatbot look cleaner

Add a new WelcomeCommand.java that will show the welcome message when user type 'Hello'
This limit the user from inputting random text that is not accepted by the chatbot

Change the return type from void to String in TaskList.java
Instead of using system println, return string so that it is easier to detect error and unit testing

Update Javadoc comment for all command files as we add a new parameter storage

Add a finally statement in Duke.java
Ensure that we write task to storage regardless of whether an exception occurred

ashleygoh1 committed:
Merge pull request #16 from ashleygoh1/A-FullCommitMessage

A full commit message

Commit:
Include assertion statements

Update GUI welcome message
Bee GUI shows welcome message when user inputs 'hello' command
This causes a blank GUI to be shown when user launches the chatbot.
This can be confusing to the user as they do not know what they are
suppose to do.

The initialise method in MainWindow now calls the dialogContainer to
create a new dialogBox with the bee image and the welcome
message to be printed out when the program is ran.

Commit:
Improve code quality

Update FindCommand.java, TodoCommand.java to restructure the code to make the
happy path (i.e. the execution path taken when everything goes well) less-nested

ashleygoh1 committed:
Merge pull request #17 from ashleygoh1/branch-A-Assertions

Include assertion statements

ashleygoh1 committed:
Merge pull request #18 from ashleygoh1/branch-A-CodeQuality

Improve code quality

Commit:
Set up Continuous Integration (CI)

Update DialogBox
Bee DialogBox is always fixed size.
This can lead to some tasks being omitted
when user input ‘list’.
This might confuse user as it could appear that
the list is incomplete or missing altogether.

Commit:
Introduce Update Command Feature

Previously, users were unable to modify the description of existing tasks,
necessitating the deletion and recreation of tasks.
A new update command allow users to modify existing task
descriptions with more convenience.

Commit:
remove gradle.yml

ashleygoh1 committed:
Merge pull request #19 from ashleygoh1/BCD-Extension

BCD extension

Commit:
A better GUI

The dialog box might not be able to fit all content
Edit the DialogBox.fxml to allow dialogbox to have infinite max height to allow texts to fit in the dialog box perfectly

Resize the profile image for user and bot
Add drop shadow to the profile images
Format text box to include round border, background colour

Commit:
Give a unique personality

Change the phrases use by bee chatbot when responding to the user

Update GUI font and colour

Commit:
Handle unexpected date

It is possible for user to input event with start date/time happening later than the end date/time
Use compareTo to compare startDate with endDate and startTime with endTime
Display error message to user

Update code quality to remove deep nesting

Commit:
Add more automated tests

Test bye, delete, find, list, update commands

ashleygoh1 committed:
Merge pull request #21 from ashleygoh1/A-BetterGui

A better GUI

ashleygoh1 committed:
Merge branch 'master' into A-Personality

ashleygoh1 committed:
Merge pull request #22 from ashleygoh1/A-Personality

Give a unique personality

ashleygoh1 committed:
Merge branch 'master' into A-MoreErrorHandling

ashleygoh1 committed:
Merge pull request #23 from ashleygoh1/A-MoreErrorHandling

Handle unexpected date

ashleygoh1 committed:
Merge branch 'master' into A-MoreTesting

ashleygoh1 committed:
Merge pull request #24 from ashleygoh1/A-MoreTesting

Add more automated tests

Commit:
User Guide

Add a User Guide detailing usage instructions for all essential features

ashleygoh1 committed:
Merge pull request #25 from ashleygoh1/A-UserGuide

User Guide

Commit:
Edit code to adhere to Checkstyle conventions and updated JUnit tests

ashleygoh1 committed:
Merge pull request #26 from ashleygoh1/A-Release

Edit code to adhere to Checkstyle conventions and updated JUnit tests

ashleygoh1 committed:
Update README.md

Commit:
edit the code based on feedback received from peer review

no error message was shown when user input unknown command
in Parser.java, use if else instead of switch case enum to show error message

program does not automatically close
in MainWindow.java, add pause transition to allow program to pause
and show the bye message before closing the program

ashleygoh1 committed:
Merge pull request #28 from ashleygoh1/bug-fix

edit the code based on feedback received from peer review
